[iis] Fix service check on the `_Total` site

olivielpeau opened the pull request:
Before 5.7.0, for `sitename == '_Total'`, we used to send a service check.
5.7.0 to 5.7.2 were always  sending a critical service check for this site name.

Go back to the previous behavior, and add a test case.

Should fix #2354

truthbk commented on tests/core/test_wmi.py:
```diff
@@ -172,9 +172,15 @@ def ExecQuery(self, query, query_language, flags):
                      "TotalConnectionAttemptsAllInstances,TotalGetRequests,TotalPostRequests,TotalHeadRequests,TotalPutRequests,TotalDeleteRequests,"  # noqa
                      "TotalOptionsRequests,TotalTraceRequests,TotalNotFoundErrors,TotalLockedErrors,TotalAnonymousUsers,TotalNonAnonymousUsers,TotalCGIRequests,"  # noqa
-                     "TotalISAPIExtensionRequests from Win32_PerfFormattedData_W3SVC_WebService WHERE ( Name = 'Failing site' ) OR ( Name = 'Default Web Site' )"):  # noqa
+                     "TotalISAPIExtensionRequests from Win32_PerfFormattedData_W3SVC_WebService WHERE ( Name = 'Failing site' ) OR ( Name = 'Default Web Site' )"):
```
`# noqa` removed here but line is still pretty long, not sure if it would generate a flake8/pep8 warning... also it was kept in line 181 below. We should at least be consistent.

olivielpeau replied:
Thanks for catching this, definitely shouldn't have been part of my changes.

truthbk commented:
Other than the nitpick, it looks good! :+1:

truthbk commented:
Let's make sure CI passes though, something weird happened to Travis.

olivielpeau commented:
Addressed your comment, thanks!

The CI fails on python 2.6, apparently because of an issue with the install of psutil, I'll try to have a look, but I don't want to spend too much time on it since we'll stop supporting python2.6 soon.

bronandrews commented:
What version what this resolved in?
Just installed latest (5.7.4) and still seeing continuous Warnings in the Collector Logs

2016-04-26 15:17:39 GMT Daylight Time | WARNING | checks.iis(iis.py:141) | When extracting metrics with WMI, found a non digit value for property 'name'.
2016-04-26 15:17:56 GMT Daylight Time | WARNING | checks.iis(iis.py:141) | When extracting metrics with WMI, found a non digit value for property 'name'.
2016-04-26 15:18:13 GMT Daylight Time | WARNING | checks.iis(iis.py:141) | When extracting metrics with WMI, found a non digit value for property 'name'.
2016-04-26 15:18:30 GMT Daylight Time | WARNING | checks.iis(iis.py:141) | When extracting metrics with WMI, found a non digit value for property 'name'.
2016-04-26 15:18:47 GMT Daylight Time | WARNING | checks.iis(iis.py:141) | When extracting metrics with WMI, found a non digit value for property 'name'.
2016-04-26 15:19:04 GMT Daylight Time | WARNING | checks.iis(iis.py:141) | When extracting metrics with WMI, found a non digit value for property 'name'.
2016-04-26 15:19:21 GMT Daylight Time | WARNING | checks.iis(iis.py:141) | When extracting metrics with WMI, found a non digit value for property 'name'.

olivielpeau commented:
Hi @bronandrews,

The changes brought by this PR were shipped with 5.7.3, and fixed the service check that's sent by the IIS check on the `_Total` website.

The warnings you see in the collector logs are false positives that are unrelated to this issue and should not have any impact on the behavior of the IIS check. We'll work on removing these warnings as they are clearly misleading.

Thanks!

bronandrews commented:
That's great - good to know there is no underlying issue. 

Many thanks for the swift response!
